src/domUtils.ts

```diff
@@ -1,9 +1,12 @@
-export const createAnchor = (href: string, target?: string) => {
+export const createAnchor = (href: string, target?: string, callback?: () => void) => {
```

Comment:
(minor) just to keep the convention of naming the params the same as the attributes, can you call it `onclick`?